File: TimeCalculator/TimeInterval.cs
namespace TimeCalculator;

public class TimeInterval
{
    private readonly Time _startTime;
    private readonly Time _endTime;

    public TimeInterval(Time startTime, Time endTime)
    {
        _startTime = startTime;
        _endTime = endTime;
    }

    public Time StartTime => _startTime;

    public Time EndTime => _endTime;

    /// <summary>
    /// Calculate the time interval in minutes.
    /// </summary>
    /// <returns></returns>
    public int GetTimeIntervalInMinutes()
    {
        return _startTime.GetTimeDifferenceInMinutes(_endTime);
    }

    /// <summary>
    /// format hh:mm ~ hh:mm
    /// </summary>
    /// <returns></returns>
    public string GetTimeIntervalString()
    {
        return $"{StartTime.GetTimeString()} ~ {EndTime.GetTimeString()}";
    }
}

File: src/time_interval.rs
use super::time::Time;

pub struct TimeInterval {
    start_time: Time,
    end_time: Time,
}

impl TimeInterval {
    pub fn new(start_time: Time, end_time: Time) -> Self {
        Self {
            start_time,
            end_time,
        }
    }

    pub fn get_start_time(&self) -> &Time {
        &self.start_time
    }

    pub fn get_end_time(&self) -> &Time {
        &self.end_time
    }

    pub fn get_time_interval_in_minutes(&self) -> u16 {
        self.start_time.get_difference_in_minutes(&self.end_time)
    }

    pub fn get_time_interval_string(&self) -> String {
        format!(
            "{} ~ {}",
            self.start_time.get_time_string(),
            self.end_time.get_time_string(),
        )
    }
}
